far/editor.cpp

```diff
@@ -318,15 +327,27 @@ void Editor::ShowEditor()
 	auto LeftPos = m_it_CurLine->GetLeftPos();
 	const Edit::ShowInfo info{ LeftPos, CurPos };
 	auto Y = m_Where.top;
+	int LineNum = m_it_TopScreen.Number() + 1;
 
-	for (auto CurPtr = m_it_TopScreen; CurPtr != Lines.end() && Y <= m_Where.bottom; ++CurPtr, ++Y)
+	for (auto CurPtr = m_it_TopScreen; CurPtr != Lines.end() && Y <= m_Where.bottom; ++CurPtr, ++Y, ++LineNum)
 	{
 		CurPtr->SetEditBeyondEnd(true);
-		CurPtr->SetPosition({ m_Where.left, Y, XX2, Y });
+		CurPtr->SetPosition({ m_Where.left + LineNumWidth, Y, XX2, Y });
 		CurPtr->SetLeftPos(LeftPos);
 		CurPtr->SetTabCurPos(CurPos);
 		CurPtr->SetEditBeyondEnd(EdOpt.CursorBeyondEOL);
 
+		// Draw line numbers if enabled
+		if (EdOpt.ShowLineNumbers && LineNumWidth > 0)
+		{
+			const auto LineNumStr = std::format(L"{:>{}}", LineNum, LineNumWidth - 1);
+			const auto LineNumColor = colors::PaletteColorToFarColor(COL_EDITORLINENUMBERS);
+			SetScreen({ m_Where.left, Y, m_Where.left + LineNumWidth - 1, Y }, L' ', LineNumColor);
```

Comment:
The area will be almost entirely overwritten by `Text` anyway.
Please change the format string to `L"{:>{}} "` (+trailing space for spacing) and remove this.